src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdater.java

```diff
@@ -0,0 +1,222 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd;
+
+import feign.FeignException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.InvalidCaseDataException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.caseupdate.CaseUpdateClient;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.caseupdate.model.response.CaseUpdateDetails;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.caseupdate.model.response.SuccessfulUpdateResponse;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.model.request.ExceptionRecord;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.config.ServiceConfigItem;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.callback.CallbackException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.callback.ProcessResult;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.payments.PaymentsPublishingException;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+
+import java.util.Map;
+
+import static com.microsoft.applicationinsights.core.dependencies.io.grpc.netty.shaded.io.netty.handler.codec.http.HttpResponseStatus.BAD_REQUEST;
+import static java.lang.String.format;
+import static java.util.Collections.emptyList;
+
+@Service
+public class CcdCaseUpdater {
+    private static final Logger log = LoggerFactory.getLogger(CcdCaseUpdater.class);
+
+    private static final String EXCEPTION_RECORD_REFERENCE = "bulkScanCaseReference";
+
+    private final CaseUpdateClient caseUpdateClient;
+    private final AuthTokenGenerator s2sTokenGenerator;
+    private final CoreCaseDataApi coreCaseDataApi;
+    private final ExceptionRecordFinalizer exceptionRecordFinalizer;
+
+    public CcdCaseUpdater(
+        CaseUpdateClient caseUpdateClient,
+        AuthTokenGenerator s2sTokenGenerator,
+        CoreCaseDataApi coreCaseDataApi,
+        ExceptionRecordFinalizer exceptionRecordFinalizer
+    ) {
+        this.caseUpdateClient = caseUpdateClient;
+        this.s2sTokenGenerator = s2sTokenGenerator;
+        this.coreCaseDataApi = coreCaseDataApi;
+        this.exceptionRecordFinalizer = exceptionRecordFinalizer;
+    }
+
+    @SuppressWarnings({"squid:S2139", "unchecked"}) // squid for exception handle + logging
+    public ProcessResult updateCase(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'updateCase(ExceptionRecord, ServiceConfigItem, boolean, String, String, CaseDetails)' has a cyclomatic complexity of 10.](https://app.codacy.com/app/HMCTS/bulk-scan-orchestrator/pullRequest?prid=4558409)